src/Bindings/ManualBindings_World.cpp

```diff
@@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
 		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
 	}
 
-	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
-	Plugin->AddResettable(task);
-	self->QueueTask(task);
+	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
+	Plugin->AddResettable(ResettableTask);
+	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
```

Comment:
Yes, lambdas still do create a class, but I was intending to convey that it is much less effort now that before, and World.h is that much smaller.

On 19 Sep 2015, at 21:55, "Mattes D" <notifications@github.com> wrote:

> In src/Bindings/ManualBindings_World.cpp:
> 
> > @@ -380,9 +378,9 @@ static int tolua_cWorld_QueueTask(lua_State * tolua_S)
> >  		return cManualBindings::lua_do_error(tolua_S, "Error in function call '#funcname#': Could not get function reference of parameter #1");
> >  	}
> >  
> > -	auto task = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
> > -	Plugin->AddResettable(task);
> > -	self->QueueTask(task);
> > +	auto ResettableTask = std::make_shared<cLuaWorldTask>(*Plugin, FnRef);
> > +	Plugin->AddResettable(ResettableTask);
> > +	self->QueueTask(std::bind(&cLuaWorldTask::Run, ResettableTask, std::placeholders::_1));
> You still have one class for each action that is to be scheduled, so there's no real improvement here - only that your construct doesn't inherit from a common predecessor.
> 
> —
> Reply to this email directly or view it on GitHub.
>